### PR TITLE
Use async to improve download performance

### PR DESCRIPTION
**Pull request by sansmoraxz**
I have observed severe slow download speed compared to native download of chrome. On investigating code I have observed small buffer size, repeated flush calls and blocking requests call which severely slows down IO Ops. This is an attempt to free up some resources and improve performance.

### CODE REVIEW

File: __init__.py [hunk 1]
```diff
@@ -778,98 +779,85 @@ async def get_directory_list(request):
...
-                            "-" * progress,
-                            " " * (50 - progress),
-                            100 * fraction,
-                        )
+                      "\r[%s%s] %d%%"
+                      % (
+                        "-" * progress,
+                        " " * (50 - progress),
+                        100 * fraction,
+                      )
```
Comment (sansmoraxz) [inline __init__.py:851]: This is kinda clunky while downloading files in parallel. Also it messes up the log file. I would suggest removing it and adding some other way to track download progress.

Comment (sansmoraxz) [reply __init__.py:840]: @hayden-fr do you wanna keep this, or should I remove it?

**Comment by hayden-fr** [comment]
Sorry, I just saw your PR now. 

Recently I refactored the project, including the download method, you can try it #38

**Comment by hayden-fr** [comment]
I'll take the time to test your method in the next few days.

If the method you provide can improve the download speed, you are welcome to resubmit your code after the new UI is approved.

**Comment by hayden-fr** [comment]
As mentioned in #35 , use aria to solve it.

Detect whether the user has aria locally. If so, we can try to use aria2 to download. If not, fall back to the default method.

**Comment by sansmoraxz** [comment]
Will test with this new update. Not sure if depending on third party solution is a good idea.

Closing this. Will send a new PR.

**Pull request closed without merging**